src/core/address_space.cpp

```diff
@@ -244,22 +274,72 @@ enum PosixPageProtection {
 struct AddressSpace::Impl {
     Impl() {
         // Allocate virtual address placeholder for our address space.
-        void* hint_address = reinterpret_cast<void*>(SYSTEM_MANAGED_MIN);
-        virtual_size = SystemSize + UserSize;
-        virtual_base = reinterpret_cast<u8*>(
-            mmap(hint_address, virtual_size, PROT_READ | PROT_WRITE,
+        system_managed_size = SystemManagedSize;
+        system_reserved_size = SystemReservedSize;
+        user_size = UserSize;
+#ifdef __APPLE__
+        system_managed_base = reinterpret_cast<u8*>(
```

Comment:
Make a constexpr variable with the common flags MAP_PRIVATE | MAP_ANONYMOUS | MAP_NORESERVE to reduce visual noise